specs/latest/2.0/index.html

```diff
@@ -3325,13 +3325,16 @@ <h2 class="no-toc">Table of contents</h2>
     </div>
 
     <p>
-        If an ESSL1 fragment shader writes to neither <code>gl_FragColor</code> nor <code>gl_FragData</code>,
-        the values of the fragment colors following shader execution are untouched. If corresponding output
-        variables are not defined in an ESSL3 fragment shader, the values of the fragment colors following
-        shader execution are untouched.
+        If fragment output variables are not defined in an ESSL3 fragment shader, the values of the
+        fragment colors following shader execution are untouched.
```

Comment:
This is ok, but requires draw-time DrawBuffer patching.

GLES3 p185:
"If an OpenGL ES Shading Language 1.00 fragment shader writes to gl_-
FragColor or gl_FragData, DrawBuffers specifies the draw buffer, if any, into
which the single fragment color defined by gl_FragColor or gl_FragData[0]
is written. If an OpenGL ES Shading Language 3.00 fragment shader writes a user-
defined varying out variable, DrawBuffers specifies a set of draw buffers into
which each of the multiple output colors defined by these variables are separately
written. If a fragment shader writes to none of gl_FragColor, gl_FragData,
nor any user-defined output variables, the values of the fragment colors following
shader execution are undefined, and may differ for each fragment color."

I think it's better if we just INVALID_OPERATION if any color draw buffer is non-null+non-NONE.

Reply:
I'm fine with the INVALID_OPERATION if a shader doesn't write all color buffers that are enabled in the framebuffer. But since it is a breaking change so we'd have to watch out for regressions, and I would like to hear other opinions on it too. Maybe that would also be better to handle in a separate commit, with tests changed in sync with the spec.

Reply:
Right, I think we'd probably break some content by generating INVALID_OP whenever /any/ color buffer is bound to an unwritten frag output. However, it should be much less common for content to write to /none/ of the color outputs. I think we'd be safe making that INVALID_OP, particularly since this is WebGL 2 only.